---
title: "Timing the first multi-gain events in Multiple Myeloma"
output:
  html_document: default
  pdf_document: default
  word_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options("scipen"=50, "digits"=4)
maxiter=2000
```

## Introduction: Molecular and absolute time in Multiple Myeloma

Most tumors and virtually all normal tissues accumulate mutations at a consistent rate, typically specific to each tissue (Alexandrov et al., Nat Gen 2015; Gerstung et al., Nature 2020; Moore et al., Nature 2021). This accumulation rate, often referred to as a biological 'clock,' is minimally influenced by the tissue's replication rate (Abascal et al. Nature 2021). The estimation of this clock relies on two distinct single base substitution (SBS) signatures: SBS1 and SBS5. Notably, in both aggressive and indolent lymphoma, as well as in normal B-cells, SBS1 and SBS5 exhibit a linear correlation with patients' age (Gerstung et al., Nature 2020; Machado et al., Nature 2020), thereby confirming the presence of the biological clock, even in the context of germinal center exposure.

Earlier studies, albeit on limited sample sizes, indicated a similar clock correlation in multiple myeloma (Rustad et al., Nat Comm 2020). Building upon these preliminary findings, our study delved into 421 whole-genome sequencing (WGS) samples from multiple myeloma patients to validate and expand upon these early observations. We demonstrate that the mutational burden represented by SBS1 and SBS5 can effectively serve as a temporal marker, aligning with findings in other cancer types."

I refined the language for clarity and coherence, ensured proper citation formatting, and provided a smoother transition between sentences. Let me know if you need further adjustments!


```{r Libraries}

packages <- c('readr', 'tidyr', "splitstackshape", "plyr", "dplyr", "ggplot2", "ggpubr", "reshape2", "magrittr", "lme4", "lmerTest", "knitr",
              "ggplot2","reshape2", "MASS", "RColorBrewer", "stringr", "deconstructSigs", "BSgenome.Hsapiens.UCSC.hg19",
              "stringi", "tibble", "pander", "RColorBrewer", "merTools")

invisible(suppressWarnings(suppressMessages(lapply(packages, library, character.only = TRUE))))
```

The uploaded file contains the contribution of each SBS signature for each sample. SBS signatures were estimated using mmsig (https://github.com/UM-Myeloma-Genomics/mmsig)

```{r}

# upload file with the mutational signature contribution for each patient
sig_age<- read.delim("timing_clock.txt")
head(sig_age)
```

Upload purity and coverage data and combine with mutational signature data frame

```{r}
coverage_purity<- read.delim("Pre_Post_gain_JCOclassification_purity_coverage_421WGS.txt")
sig_age[is.na(sig_age)]<-""
coverage_purity[is.na(coverage_purity)]<-""
sig_age_cov<- unique(merge(sig_age, coverage_purity, by="sample"))
sig_age_cov$age_at_sample_collection<- as.numeric(as.character(sig_age_cov$age_at_sample_collection)) # age at sample collection
sig_age_cov$abs_clock<- as.numeric(as.character(sig_age_cov$abs_clock)) # number of SBS1 and SBS5 mutations
sig_plot<- sig_age_cov[order(sig_age_cov$apo_rr_state),]
sig_plot<- unique(sig_plot[!is.na(sig_plot$age_at_sample_collection),])
```

Similarly to Gerstung et al. we assessed if the the relationship between the mutational burden of SBS1 and SBS5 and the age at sample collection could be explained by a linear model with the intercept constrained to zero. 

```{r}
### Interception constrained to zero as Gerstung et al. Nature 2020

summary(lm(sig_plot$abs_clock~ 
             0+sig_plot$age_at_sample_collection))

```
Then, the regression model was adjusted for hyper-APOBEC activity, the presence of melphalan mutational signatures, and disease stage (relapse vs. newly diagnosed).

```{r}
summary(lm((sig_plot$abs_clock~sig_plot$age_at_sample_collection+sig_plot$apo_rr_state)))
```


Visualize the correlation between SBS1 and SBS5 and age in multiple myeloma

```{r, warning=FALSE}
### ALL SBS5 and SBS1
par(xpd=F, mar=c(5,5,2,5))
plot(x=sig_plot$age_at_sample_collection,pch=21,bg=sig_plot$apobec_hyper,
     y=sig_plot$abs_clock, xlim=c(0, 100), ylim=c(0,10000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
axis(side = 2, at = seq(0,10000, by=2000),labels = seq(0,10000, by=2000), las=2, cex.axis=1.5, lwd=1.5)
axis(side = 1, at = seq(0,100, by=20),labels = seq(0,100, by=20), las=1, cex.axis=1.5, lwd=1.5)
abline(lm(sig_plot$abs_clock~sig_plot$age_at_sample_collection+sig_plot$apobec_hyper))
```

We further investigate the correlation between SBS1/SBS5 and age while correcting for potential confounders such as coverage and purity Additionally, the presence of hyperAPOBEC was tested, as we previously demonstrated that, similar to other hypermutated tumors (e.g., Melanoma), the clock-like signal becomes unreliable due to the interbreeding of the mutations process causing hypermutation.


```{r}
### coverage and purity don;t affect the correlation

sig_plot$Coverage.TUMOR<- as.numeric(as.character(sig_plot$Coverage.TUMOR))
sig_plot$Purity<- as.numeric(as.character(sig_plot$Purity))

summary(lm(sig_plot$abs_clock~ 
             sig_plot$age_at_sample_collection + sig_plot$apo_rr_state+
               sig_plot$Coverage.TUMOR/sig_plot$Purity))

```

Here we estimate the residual for each patient. Patients with an excess of residual.
To be consistent with recent litterature we decide to use the model with the intercept constrain to 0. 

```{r, warning=F}
model_res<- (lm(sig_plot$abs_clock~ 0+sig_plot$age_at_sample_collection))
residuals_vec <- residuals(model_res)

# Calculate absolute values of residuals
abs_residuals <- abs(residuals_vec)
sig_plot$residual<- abs_residuals
sig_plot<- sig_plot[order(sig_plot$residual),]
plot((sig_plot$residual), col=sig_plot$apobec_hyper,pch=16, ylim=c(0,10000), ylab="Residual")
```

The number of cases that do not fit the linear model will be excluded from the absolute timing estimation analysis.

```{r, warning=F}
cutff_model<-1900
length(sig_plot$apobec_hyper[sig_plot$residual>cutff_model]) 
```


```{r, warning=F}

sig_plot$residual_code<- 0
sig_plot$residual_code[sig_plot$residual>cutff_model]<- 1

par(xpd=F, mar=c(5,5,2,5))
# [sig_age2$abs_clock<8000,]
plot(sig_plot$age_at_sample_collection[sig_plot$residual>cutff_model],sig_plot$abs_clock[sig_plot$residual>cutff_model],
     pch=16, col=alpha(sig_plot$apobec_hyper[sig_plot$residual>cutff_model], 0.6),
     xlim=c(0, 100), ylim=c(0,10000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
par(new=T)
plot(sig_plot$age_at_sample_collection[sig_plot$residual>cutff_model],sig_plot$abs_clock[sig_plot$residual>cutff_model],
     pch=21, bg = NA, col="grey",
     xlim=c(0, 100), ylim=c(0,10000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")

par(new=T)
plot(sig_plot$age_at_sample_collection[sig_plot$residual<cutff_model],pch=21,bg=sig_plot$apobec_hyper,
     sig_plot$abs_clock[sig_plot$residual<cutff_model], xlim=c(0, 100), ylim=c(0,10000), las=2,
     ylab="", xlab="",  bty="n")
abline(lm(sig_plot$abs_clock[sig_plot$residual<cutff_model]~0+sig_plot$age_at_sample_collection
          [sig_plot$residual<cutff_model]))


```

## Calculate impact of IGH translocations


```{r}
### Interception not constrain to zero and corrected for coverage, purity, and disease state and hyperAPOBEC


### ALL SBS5 and SBS1
par(xpd=F, mar=c(5,5,2,5))


sig_age_plot<- sig_age_cov[order(sig_age_cov$Ig),]

sig_age_plot$color_igh<-"coral"
sig_age_plot$color_igh[sig_age_plot$Ig!="no"]<-"grey50"
sig_age_plot$age_at_sample_collection<- as.numeric(as.character(sig_age_plot$age_at_sample_collection))
# sig_age_plot<- sig_age_plot[sig_age_plot$abs_clock<8000,]
plot(sig_age_plot$age_at_sample_collection[sig_age_plot$Ig=="no"],pch=21,bg=sig_age_plot$color_igh[sig_age_plot$Ig=="no"],
     sig_age_plot$abs_clock[sig_age_plot$Ig=="no"], xlim=c(0, 100), ylim=c(0,12000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
par(new=T)
plot(sig_age_plot$age_at_sample_collection[sig_age_plot$Ig!="no"],pch=21,bg=sig_age_plot$color_igh[sig_age_plot$Ig!="no"],
     sig_age_plot$abs_clock[sig_age_plot$Ig!="no"], xlim=c(0, 100), ylim=c(0,12000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
axis(side = 2, at = seq(0,10000, by=2000),labels = seq(0,10000, by=2000), las=2, cex.axis=1.5, lwd=1.5)
axis(side = 1, at = seq(0,100, by=20),labels = seq(0,100, by=20), las=1, cex.axis=1.5, lwd=1.5)
abline(lm(sig_age_plot$abs_clock[sig_age_plot$Ig!="no"]~sig_age_plot$age_at_sample_collection[sig_age_plot$Ig!="no"]))
abline(lm(sig_age_plot$abs_clock[sig_age_plot$Ig=="no"]~sig_age_plot$age_at_sample_collection[sig_age_plot$Ig=="no"]), col="red")

```

## Myeloma and other B-cell tumors

Comparing SBS1 and SBS5 mutation rate between multiple myeloma and normal B-cell. Normal B-cell data were taken from Machado et al. Nature 2022. Because Machado et al. used single cell expansion, to make the comparison more reliable we only considered clonal mutations in multiple myeloma WGS (i.e. the most recent common ancestor).

```{r}

pcawgs<- read.delim("PCAWG_sigProfiler_SBS_signatures_in_samples.csv", sep=",")
pcawgs_lymph<-pcawgs[pcawgs$Cancer.Types %in% c("Lymph-BNHL"    ,      "Lymph-CLL" ),]

age_cpag<-read.delim("PCAWG7_age_information.txt")
colnames(age_cpag)<-c("Sample.Names","age")
pcag<- merge(age_cpag, pcawgs_lymph, by="Sample.Names")
pcag$clock<-pcag$SBS1+pcag$SBS5
pcag<-pcag[pcag$clock<8000,] ## remove hypermutated
pcag<-pcag[-grep("CLL", pcag$Cancer.Types),] ## remove CLL

sig_plot<- sig_plot[sig_plot$abs_clock<8000,] ## remove hypermutated

par(xpd=F, mar=c(5,5,2,5))
plot(sig_plot$age_at_sample_collection,pch=21,bg="dodgerblue",
     sig_plot$abs_clock, xlim=c(0, 100), ylim=c(0,10000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")

summary(lm((sig_plot$abs_clock~sig_plot$age_at_sample_collection+sig_plot$stage_col+sig_plot$apobec_hyper)))
abline(lm(sig_plot$abs_clock~sig_plot$age_at_sample_collection+sig_plot$apobec_hyper), col="dodgerblue4")

axis(side = 2, at = seq(0,10000, by=2000),labels = seq(0,10000, by=2000), las=2, cex.axis=1.5, lwd=1.5)
axis(side = 1, at = seq(0,100, by=20),labels = seq(0,100, by=20), las=1, cex.axis=1.5, lwd=1.5)
par(new=T)
plot(x=pcag$age,pch=21,bg="olivedrab3",
     y=pcag$clock, xlim=c(0, 100), ylim=c(0,10000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n", col="black")
abline(lm(pcag$clock~pcag$age), col="forestgreen")


```

```{r}
summary(lm(pcag$clock~pcag$age))

```

```{r}
summary(lm(pcag$clock~0+pcag$age)) # constrain intercept like Gerstung et al. Nature 2020

```

```{r}
machado<- read.delim("machado_mmsig.txt")
machado$id <- as.factor(machado$id)
collapsed_df <- aggregate(abs_clock ~ id, data = machado, FUN = median, na.rm = TRUE)
collapsed_df <- merge(collapsed_df, machado[!duplicated(machado$id), c("id", "Age", "CellType", "Cell.type2", "Tissue")], by = "id")
collapsed_df2<- collapsed_df[collapsed_df$Cell.type2=="Memory B",]


clonal_sig2<- read.delim("CLONAL_SBS.txt",sep="\t")


par(xpd=F, mar=c(5,5,2,5))
plot(clonal_sig2$age_at_sample_collection,pch=21,bg=clonal_sig2$apobec_hyper,
     clonal_sig2$abs_clock, xlim=c(0, 100), ylim=c(0,6000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
axis(side = 2, at = seq(0,6000, by=2000),labels = seq(0,6000, by=2000), las=2, cex.axis=1.5, lwd=1.5)
axis(side = 1, at = seq(0,100, by=20),labels = seq(0,100, by=20), las=1, cex.axis=1.5, lwd=1.5)
par(new=T)
machado2<-machado[machado$CellType == "B Memory",]
plot(machado$Age, machado$abs_clock, pch=21,bg="gold3",
     xlim=c(0, 100), ylim=c(0,6000), las=2,
     yaxt="n", ylab="", xlab="", xaxt="n", bty="n")
abline(lm(machado$abs_clock~0+machado$Age),col = "black")
abline(lm(clonal_sig2$abs_clock~0+clonal_sig2$age_at_sample_collection),col = "red")
summary(lm(machado2$abs_clock~0+machado2$Age))
summary(lm(clonal_sig2$abs_clock~clonal_sig2$age_at_sample_collection+clonal_sig2$apobec_hyper))
legend("toprigh", legend = c("Normal B-cells", "MM"), pch = "-", col=c("black","red"))
```

Compare the linear model from Machado et al. Nature 2020 B-cell SBS-SBS5 mutation rate and the one observed in multiple myeloma

```{r}
model1 <- lm(machado2$abs_clock~0+machado2$Age)
model2 <- lm(clonal_sig2$abs_clock ~0+clonal_sig2$age_at_sample_collection+clonal_sig2$apobec_hyper)
coef_model1 <- coef(summary(model1))["machado2$Age", "Estimate"]
coef_model2 <- coef(summary(model2))["clonal_sig2$age_at_sample_collection", "Estimate"]
se_model1 <- coef(summary(model1))["machado2$Age", "Std. Error"]
se_model2 <- coef(summary(model2))["clonal_sig2$age_at_sample_collection", "Std. Error"]

# Calculate the t-value
t_value <- (coef_model1 - coef_model2) / sqrt(se_model1^2 + se_model2^2)

# Calculate the degrees of freedom
df <- sum(model1$df.residual, model2$df.residual)

# Perform a t-test
p_value <- 2 * pt(abs(t_value), df = df, lower.tail = FALSE)
print(p_value)

```

